### PR TITLE
 Prototype of moving from edition based code to a separate module

### PR DESCRIPTION
**Pull request by MishaDemianenko**
This prototype should show possibility of great simplification
how dependencies are handled between modules and how
edition can be entirely build without any extension and
service loading.

### CODE REVIEW

**Comment by MishaDemianenko** [comment]
@tinwelint would be nice to hear wdyt?

File: community/neo4j/src/main/java/org/neo4j/platform/GraphDatabaseFacadeFactory.java [hunk 1]
```diff
@@ -88,25 +84,6 @@
...
-
-        @Internal
-        public static final Setting<String> tracer =
-                setting( "unsupported.dbms.tracer", STRING, Settings.NO_DEFAULT );
-
-        @Internal
-        public static final Setting<String> editionName =
-                setting( "unsupported.dbms.edition", STRING, Edition.unknown.toString() );
-    }
-
```
Comment (chrisvest) [inline community/neo4j/src/main/java/org/neo4j/platform/GraphDatabaseFacadeFactory.java:109]: I like that this is moved to GraphDatabaseSettings.

**Comment by chrisvest** [comment]
I think this opens up for a discussion about where the split is between PlatformModule and EditionModule. In a multi-tenant world, would all tenants be forced to use the same edition, or will you want to start tenants in different editions? It is complicated by our currently distinguishing CC & HA as editions as well. I could imagine running both CC and EE at the same time in different tenants, for instance. Some things that are in PlatformModule also look like they really belong in EditionModule, and vice versa.

It doesn't look like very much actually changed, other than moving files around. So I'm not sure what's simpler, and what the "without service loading" refers to.

**Comment by tinwelint** [comment]
I get lost in the amount of changes... I'm sure I'm missing the small and obvious core change in this PR, but what is it?

**Pull request closed without merging**